tests/ga/test_extension.py

```diff
@@ -2308,34 +2343,149 @@ def test_failed_env_variables_should_be_set_from_within_extension_commands(self,
+            self.assertTrue(ExtCommandEnvVariable.ExtensionPath in install_kwargs['message'] and
+                            ExtCommandEnvVariable.ExtensionVersion in install_kwargs['message'])
+            self.assertNotIn(ExtCommandEnvVariable.DisableReturnCode, install_kwargs['message'])
+
+    @patch('time.sleep', side_effect=lambda _: mock_sleep(0.001))
+    def test_correct_exit_code_should_set_on_disable_cmd_failure(self, _):
+        test_env_file_name = "test_env.sh"
+        test_failure_file_name = "test_fail.sh"
+        # update_file_name = test_env_file_name + " -update"
+        old_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance('foo', '1.0.0', handler={
+            "disableCommand": test_failure_file_name,
+            "uninstallCommand": test_failure_file_name})
+        new_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance(
+            'foo', '1.0.1',
+            handler={"updateCommand": test_env_file_name,
+                     "updateMode": "UpdateWithoutInstall"},
+            continue_on_update_failure=True
+        )
+
+        exit_code = 150
+        error_test_file = """
+                    exit %s
+                    """ % exit_code
+
+        test_env_file = """
+            printenv | grep AZURE_
+            """
+
+        self.create_script(file_name=test_env_file_name, contents=test_env_file,
+                           file_path=os.path.join(new_handler_i.get_base_dir(), test_env_file_name))
+        self.create_script(file_name=test_failure_file_name, contents=error_test_file,
+                           file_path=os.path.join(old_handler_i.get_base_dir(), test_failure_file_name))
+
+        with patch.object(new_handler_i, 'report_event', autospec=True) as mock_report:
+
+            uninstall_rc = ExtHandlersHandler._update_extension_handler_and_return_if_failed(old_handler_i, new_handler_i)
+            _, kwargs = mock_report.call_args
+
+            self.assertEqual(exit_code, uninstall_rc)
+            self.assertIn("%s=%s" % (ExtCommandEnvVariable.DisableReturnCode, exit_code), kwargs['message'])
+
+    @patch('time.sleep', side_effect=lambda _: mock_sleep(0.0001))
+    def test_timeout_code_should_set_on_cmd_timeout(self, _):
+        test_env_file_name = "test_env.sh"
+        test_failure_file_name = "test_fail.sh"
+        old_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance('foo', '1.0.0', handler={
+            "disableCommand": test_failure_file_name,
+            "uninstallCommand": test_failure_file_name})
+        new_handler_i = TestExtensionUpdateOnFailure._get_ext_handler_instance(
+            'foo', '1.0.1',
+            handler={"updateCommand": test_env_file_name + " -u", "installCommand": test_env_file_name + " -i"},
+            continue_on_update_failure=True
+        )
+
+        exit_code = 156
+        error_test_file = """
+            sleep 1m
```

Comment:
Will this actually sleep for 1 minute? If yes, can we make it shorter?

Reply:
the script will sleep for 1m but our agent will throw a timeout exception long before that, that is the main test :)